### PR TITLE
chore: component render perf tracking

### PR DESCRIPTION
**Pull request by TheSisb**
WIP tracking component render perf.

### CODE REVIEW

File: .storybook/RenderPerformanceProfiler.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import isChromatic from 'chromatic/isChromatic';
+import {version} from '../packages/paste-core/core-bundle/package.json';
+
+const trackRenderPerformance = (data) => {
+  const commitSha = process.env.STORYBOOK_GITHUB_SHA ? process.env.STORYBOOK_GITHUB_SHA : 'localdev';
+
+  // TODO: replace with https://paste-data.netlify.app/.netlify/functions/create-component-perf-metric
```
Comment (shleewhite) [inline .storybook/RenderPerformanceProfiler.js:8]: Why is this TODO?

File: .storybook/preview.js [hunk 2]
```diff
@@ -54,16 +54,18 @@ export const decorators = [
       default:
       case 'default':
         return (
-          <Theme.Provider theme={theme} disableAnimations={isChromatic()}>
+          <RenderPerformanceProfiler id={context.id} kind={context.kind} view="default">
```
Comment (shleewhite) [inline .storybook/preview.js:57]: Do we expect the view to affect the components' perf?

File: .storybook/main.js [hunk 3]
```diff
@@ -45,6 +45,15 @@ module.exports = {
     config.resolve.plugins =
       config.resolve.plugins == null ? customPlugins : [...config.resolve.plugins, ...customPlugins];
 
+    // Need to custom alias react-dom and scheduler for component profiling in production
+    // mode. Without doing so, no React profiling data can be extracted from stories
+    // When they are deployed.
+    const customAlias = {
```
[the remark this answers is not in the document]
Comment (SiTaggart) [reply .storybook/main.js:51]: We need to make it a blanket change. By default its conditional, if you use the profiler it'll use it whilst in dev mode, and remove it when compiling production code. 

Chromatic runs storybook in production mode, so we have to force it to use the profile in production mode to get the perf numbers

File: .storybook/preview.js (hunk 2 above)
Comment (SiTaggart) [reply .storybook/preview.js:57]: Potentially yeah. The mount time will be different if you just displaying the story, or if you're displaying a story side by side. It's a duplicate effectively

File: .storybook/RenderPerformanceProfiler.js (hunk 1 above)
Comment (SiTaggart) [reply .storybook/RenderPerformanceProfiler.js:8]: I need to merge the other PR in the other repo to get that url working. It's running off of a preview url right now.

Comment (shleewhite) [reply .storybook/RenderPerformanceProfiler.js:8]: 👍🏽

**Review by shleewhite — APPROVED** [review]
Looks good to me!

File: .storybook/RenderPerformanceProfiler.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (TheSisb) [reply .storybook/RenderPerformanceProfiler.js:1]: This is run by storybook. I'm not sure there's an easy way to have this be TS :(

**Review by TheSisb — COMMENTED** [review]
I can't approve my own PR with Github UI. Written approval of changes made.

**Review by zahnster — APPROVED** [review]
:doit:

File: .storybook/RenderPerformanceProfiler.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (SiTaggart) [reply .storybook/RenderPerformanceProfiler.js:1]: I think I tried, but Storybook is JS and mixed files are weird.

**Review by SiTaggart — APPROVED** [review]

**Pull request merged**